library/std/src/sys/windows/fs.rs

```diff
@@ -390,10 +395,15 @@ impl File {
             attr.file_size = info.AllocationSize as u64;
             attr.number_of_links = Some(info.NumberOfLinks);
             if attr.file_type().is_reparse_point() {
-                let mut b =
-                    Align8([MaybeUninit::<u8>::uninit(); c::MAXIMUM_REPARSE_DATA_BUFFER_SIZE]);
-                if let Ok((_, buf)) = self.reparse_point(&mut b) {
-                    attr.reparse_tag = (*buf).ReparseTag;
+                let mut attr_tag: c::FILE_ATTRIBUTE_TAG_INFO = mem::zeroed();
+                cvt(c::GetFileInformationByHandleEx(
+                    self.handle.as_raw_handle(),
+                    c::FileAttributeTagInfo,
+                    ptr::addr_of_mut!(attr_tag).cast(),
+                    mem::size_of::<c::FILE_ATTRIBUTE_TAG_INFO>().try_into().unwrap(),
```

Comment:
Kind of goofy to do this here, but I guess I'm not opposed since the unwrap should vanish and I would generally like us to be more careful about a usize->dword.

Reply:
Yeah. I was thinking of our conversation here and noticed it codegened fine. But it is weird. I do think it'd ultimately be better to have a proper wrapper for these cases that can statically assert the size fits into a u32. And also something for the cases that could possibly fail.